src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -0,0 +1,311 @@
+        internal static void AdvanceStream(Stream archiveStream, long bytesToDiscard)
+        {
+            if (archiveStream.CanSeek)
+            {
+                archiveStream.Position += bytesToDiscard;
+            }
+            else if (bytesToDiscard > 0)
+            {
+                byte[] buffer = ArrayPool<byte>.Shared.Rent(minimumLength: MaxBufferLength);
+                while (bytesToDiscard > 0)
+                {
+                    int currentLengthToRead = (int)Math.Min(MaxBufferLength, bytesToDiscard);
+                    if (archiveStream.Read(buffer.AsSpan(0, currentLengthToRead)) != currentLengthToRead)
+                    {
+                        throw new EndOfStreamException();
+                    }
+                    bytesToDiscard -= currentLengthToRead;
+                }
+                ArrayPool<byte>.Shared.Return(buffer);
+            }
+        }
+
+        // Helps copy a specific number of bytes from one stream into another.
+        internal static void CopyBytes(Stream origin, Stream destination, long bytesToCopy)
+        {
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(minimumLength: MaxBufferLength);
+            while (bytesToCopy > 0)
+            {
+                int currentLengthToRead = (int)Math.Min(MaxBufferLength, bytesToCopy);
+                if (origin.Read(buffer.AsSpan(0, currentLengthToRead)) != currentLengthToRead)
+                {
+                    throw new EndOfStreamException();
+                }
+                destination.Write(buffer.AsSpan(0, currentLengthToRead));
+                bytesToCopy -= currentLengthToRead;
+            }
+            ArrayPool<byte>.Shared.Return(buffer);
+        }
+
+        // Returns the number of bytes until the next multiple of the record size.
+        internal static int CalculatePadding(long size)
+        {
+            long ceilingMultipleOfRecordSize = ((RecordSize - 1) | (size - 1)) + 1;
+            int padding = (int)(ceilingMultipleOfRecordSize - size);
+            return padding;
+        }
+
+        // Returns the specified 8-base number as a 10-base number.
+        internal static int ConvertDecimalToOctal(int value)
+        {
+            int multiplier = 1;
+            int accum = value;
+            int actual = 0;
+            while (accum != 0)
+            {
+                actual += (accum % 8) * multiplier;
+                accum /= 8;
+                multiplier *= 10;
+            }
+            return actual;
+        }
+
+        // Returns the specified 10-base number as an 8-base number.
+        internal static long ConvertDecimalToOctal(long value)
+        {
+            long multiplier = 1;
+            long accum = value;
+            long actual = 0;
+            while (accum != 0)
+            {
+                actual += (accum % 8) * multiplier;
+                accum /= 8;
+                multiplier *= 10;
+            }
+            return actual;
+        }
```

Comment:
This is very strange to me.  The base is only relevant when rendering the number as a string; these are actually changing the numerical value so that when rendered as base 10 it produces something that looks like the original value when rendered as base 8... we should just render it as base 8 to begin with if that's what we need.